src/saveload/saveload.cpp

```diff
@@ -2863,18 +2863,17 @@ static std::pair<const SaveLoadFormat &, uint8_t> GetSavegameFormat(const std::s
 		for (const auto &slf : _saveload_formats) {
 			if (slf.init_write != nullptr && name.compare(slf.name) == 0) {
 				if (has_comp_level) {
-					const std::string complevel(full_name, separator + 1);
+					auto complevel = std::string_view(full_name).substr(separator + 1);
```

Comment:
You might want to consider changing the parameter to `GetSavegameFormat` to `std::string_view`.